### PR TITLE
Feature/indexer

### PR DESCRIPTION
**Pull request by Shikkic**
Creates basic indexer functionality.

Currently creates two JSON dumps for now, the first is successfully formed go packages, the second is packages that don't exist on Github anymore.

I average that the run time takes about 2 mins

### CODE REVIEW

File: indexer/main.go [hunk 1]
```diff
@@ -2,59 +2,166 @@ package main
...
+	"sync"
+	"time"
 )
 
+type GoPackage struct {
+	Description    string
+	GitHubURL      string
+	GoDocURL       string
+	IndexTime      string
+	HttpStatusCode int
```
Comment (Shikkic) [inline indexer/main.go]: Do we want to include packages that are 404 on github? If so we can store an identifier here? Might be nice to keep them in cassandra/elastisearch so we can tell if a package no longer exists without having to make a request

Comment (skeswa) [inline indexer/main.go]: That's a great idea. Let's throw it in.

File: indexer/main.go [hunk 2]
```diff
@@ -2,59 +2,166 @@ package main
...
+				goPackage.Description = childDescription
 			}
 		})
 
-		if strings.Contains(githubURL, "github.com") {
-			// TODO build go doc url
-			//goDocUrl =
+		// Only continue if this goPackage contains a GitHub URL
+		// TODO check to make sure github.com is the prefix
+		gitHubUrlTokens := strings.Split(goPackage.GitHubURL, "/")
```
Comment (skeswa) [inline indexer/main.go]: You use "URL" in the variable name here and "URL" elsewhere. Tsk tsk.

Comment (Shikkic) [inline indexer/main.go]: Damnnnn Daniel, you're right

**Pull request merged**